packages/framework/src/Model/Product/Pricing/ProductPriceCalculationForCustomerUser.php

```diff
@@ -30,7 +30,7 @@ public function __construct(
      * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
      * @return \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice
      */
-    public function calculatePriceForCurrentUser(Product $product)
+    public function calculatePriceForCurrentUser(Product $product): ProductPrice
```

Comment:
Here you use final class return typehint so this method cannot be changed. Should ProductPrice implement ProductPriceInterface?